Dockerfile

```diff
@@ -45,26 +47,28 @@ RUN buildDeps=( \
 			| sort -u \
 		)" \
 	&& readarray runDepsArr <<< "$runDeps" \
-	&& apk add --virtual .run-deps "${runDepsArr[@]}" gnutls-utils iptables libnl3 readline libseccomp-dev lz4-dev \
+	&& apk add --virtual .run-deps "${runDepsArr[@]}" gnutls-utils iptables libnl3 readline libseccomp-dev lz4-dev gettext-envsubst \
 	&& apk del .build-deps \
 	&& rm -rf /var/cache/apk/*
 
 # Setup config
 COPY routes.txt /tmp/
+
+# hadolint ignore=SC2016
```

Comment:
Is there a reason not to double quote, therefore prevent ignoring this hadolint rule?

Reply:
If we double-quote the lines with shell variables, then they will be interpolated during the image build, and therefore be hardcoded in the config.

By single-quoting, the variables are not replaced in the build, but by envsubst in the entrypoint, so users can do something like `docker run -e OC_IPV4_NETWORK="192.168.90.0" ...`. With double quotes, only the routes/iptables rule will honor the env var, but the config will still have the network defined during the build.

Reply:
AFAICS this is only to change /tmp/ocserv.conf, I fail to see how can this affect `envsubst` in in runtime?

Reply:
The entrypoint now has this line:
https://github.com/aminvakil/docker-ocserv/blob/ed9001d1b791d6693713c624767048d661d6318d/docker-entrypoint.sh#L75

which will take the processed config, but which still contains the env vars like `${OC_IPV4_NETWORK}`, and then replace those env vars to actual values, and write to `/etc/ocserv/ocserv.conf`. I hope that clears things up; happy to explain further, if you'd like.

Reply:
Yes, I understood this, my question is why couldn't we change single quotes to double quotes in Dockerfile?

Reply:
Below is an example of the difference.

Single quotes:
```
$ docker run --entrypoint bash -ti oc -c "grep ^ipv4-network /tmp/ocserv.conf"
ipv4-network = ${OC_IPV4_NETWORK}
```

Double quotes:
```
$ docker run --entrypoint bash -ti oc -c "grep ^ipv4-network /tmp/ocserv.conf"
ipv4-network = 192.168.99.0
```

With the former, if someone runs it like `docker run -e OC_IPV4_NETWORK="192.168.90.0" ...`, envsubst will update the config correctly in the entrypoint, and things will work as expected. With double quotes, `ipv4-network = 192.168.99.0` is hardcoded in the config at this point, and the network changes will only apply to the routes and the iptables rule, so the networks will not match.

I hope that helps explain it.